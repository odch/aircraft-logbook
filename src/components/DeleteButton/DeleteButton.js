import React from 'react'
import PropTypes from 'prop-types'
import Button from '@material-ui/core/Button'
import DeleteIcon from '@material-ui/icons/Delete'
import CircularProgress from '@material-ui/core/CircularProgress'

const DeleteButton = ({
  onClick,
  label,
  type,
  color,
  variant,
  disabled,
  inProgress,
  ...other
}) => (
  <Button
    {...other}
    onClick={onClick}
    type={type}
    color={color}
    variant={variant}
    disabled={disabled}
  >
    {inProgress ? <CircularProgress size={16} /> : <DeleteIcon />}
    {label}
  </Button>
)

DeleteButton.propTypes = {
  onClick: PropTypes.func,
  label: PropTypes.string,
  type: PropTypes.string,
  color: PropTypes.string,
  variant: PropTypes.string,
  disabled: PropTypes.bool,
  inProgress: PropTypes.bool
}

export default DeleteButton
