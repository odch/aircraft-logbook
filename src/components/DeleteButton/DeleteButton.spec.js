import React from 'react'
import DeleteButton from './DeleteButton'
import renderer from 'react-test-renderer'

describe('components', () => {
  describe('DeleteButton', () => {
    it('renders correctly', () => {
      const tree = renderer
        .create(<DeleteButton label="Delete foobar" />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
