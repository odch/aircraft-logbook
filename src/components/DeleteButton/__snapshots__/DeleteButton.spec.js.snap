// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components DeleteButton renders correctly 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
      />
    </svg>
    Delete foobar
  </span>
</button>
`;
